### PR TITLE
Handle scopes with same name in separate functions

### PR DESCRIPTION
**Pull request by vinok88**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/9550.
symbol owner name to the scope name and generated scope functions.

### CODE REVIEW

**Comment by hasithaa** [comment]
@vinok88 Are we going to have compensated statement or join this concept with transaction statement?

**Comment by vinok88** [comment]
@hasithaa joining with transactions syntax is not yet finalized. This PR is to fix a bug in existing implementation.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java [hunk 1]
```diff
@@ -2047,4 +2068,19 @@ private void validateDocAttribute(BLangDocumentationAttribute attribute) {
         }
         attribute.type = attributeSymbol.type;
     }
+
+    private void updateChildScopeNames(Stack<String> childScopes) {
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java]: Can we write this method with, 
childScopes.replaceAll(s -> scopesMap.get(s));

Comment (vinok88) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java]: Fixed. Thanks.

**Comment by hasithaa** [comment]
Shall we add a test case for this scenario?

**Comment by vinok88** [comment]
Test case added.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java [hunk 2]
```diff
@@ -1496,15 +1502,27 @@ public void visit(BLangScope scopeNode) {
         symbolEnter.defineNode(scopeNode.compensationFunction.function, env);
         typeChecker.checkExpr(scopeNode.compensationFunction, env);
         symbolEnter.defineNode(scopeNode, env);
+        BSymbol symbol = symResolver.lookupSymbol(env, names.fromString(scopeNode.name.getValue()), SymTag.SCOPE);
+        //Add to map to be used by parent scopes
+        String uniqueName = getUniqueScopeName(symbol.pkgID, symbol.owner, scopeNode.name.getValue());
+        scopesMap.put(scopeNode.name.getValue(), uniqueName);
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java]: When is this map gets cleaned? Seems like data is accumulating. Seems like this logic won't work with two functions which have a componsate statements.

**Comment by vinok88** [comment]
Closing this PR, since we identified a better way of fixing this.

**Pull request closed without merging**